lapce-ui/src/editor/tab_header_content.rs

```diff
@@ -248,6 +249,79 @@ impl LapceEditorTabHeaderContent {
             ));
         }
     }
+
+    fn get_effective_path(
+        &mut self,
+        workspace_path: Option<PathBuf>,
+        file_path: &PathBuf,
+    ) -> String {
+        let workspace_path = match workspace_path {
+            Some(p) => p,
+            None => {
+                let mut file_path_str = match file_path.to_str() {
```

Comment:
Minor: `file_path.to_str().unwrap_or("")` (then call `.to_string` on that)

Reply:
You could also just use https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.to_string_lossy

Reply:
Minor: In Rust, shadowing names (so using file_path for the String version too) is normal when they are not needed anymore (like your converting it into a String).